modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java

```diff
@@ -0,0 +1,580 @@
+        jsonWriter = new ObjectMapper(new JsonFactory());
+        claimsResponse = jsonWriter.readValue(expectedResponse, Claims.class);
+
+        expectedResponse = readResource("get-roles-response.json");
+        jsonWriter = new ObjectMapper(new JsonFactory());
+        rolesResponse = jsonWriter.readValue(expectedResponse, Roles.class);
+
+        expectedResponse = readResource("get-jit-response.json");
+        jsonWriter = new ObjectMapper(new JsonFactory());
+        justInTimeProvisioningResponse = jsonWriter.readValue(expectedResponse, JustInTimeProvisioning.class);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testConclude() {
+
+        super.conclude();
+    }
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit() {
+
+        RestAssured.basePath = basePath;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void testFinish() {
+
+        RestAssured.basePath = StringUtils.EMPTY;
+    }
+
+    @DataProvider(name = "restAPIUserConfigProvider")
+    public static Object[][] restAPIUserConfigProvider() {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN},
+                {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @Test
+    public void testListMetaFederatedAuthenticators() throws Exception {
+
+        String key1Identifier = "find{ it.authenticatorId == 'T2ZmaWNlMzY1QXV0aGVudGljYXRvcg' }.";
+        String key2Identifier = "find{ it.authenticatorId == 'VHdpdHRlckF1dGhlbnRpY2F0b3I' }.";
+        String key3Identifier = "find{ it.authenticatorId == 'RmFjZWJvb2tBdXRoZW50aWNhdG9y' }.";
+
+        String key4Identifier = "find{ it.authenticatorId == 'R29vZ2xlT0lEQ0F1dGhlbnRpY2F0b3I' }.";
+        String key5Identifier = "find{ it.authenticatorId == 'TWljcm9zb2Z0V2luZG93c0xpdmVBdXRoZW50aWNhdG9y' }.";
+        String key6Identifier = "find{ it.authenticatorId == 'UGFzc2l2ZVNUU0F1dGhlbnRpY2F0b3I' }.";
+        String key7Identifier = "find{ it.authenticatorId == 'WWFob29PQXV0aDJBdXRoZW50aWNhdG9y' }.";
+        String key8Identifier = "find{ it.authenticatorId == 'SVdBS2VyYmVyb3NBdXRoZW50aWNhdG9y' }.";
+        String key9Identifier = "find{ it.authenticatorId == 'U0FNTFNTT0F1dGhlbnRpY2F0b3I' }.";
+        String key10Identifier = "find{ it.authenticatorId == 'T3BlbklEQ29ubmVjdEF1dGhlbnRpY2F0b3I' }.";
+        String key11Identifier = "find{ it.authenticatorId == 'RW1haWxPVFA' }.";
+        String key12Identifier = "find{ it.authenticatorId == 'U01TT1RQ' }.";
+
+        Response response = getResponseOfGet(IDP_API_BASE_PATH + PATH_SEPARATOR + META_FEDERATED_AUTHENTICATORS_PATH);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body(key1Identifier + "name", equalTo("Office365Authenticator"))
+                .body(key1Identifier + "self", equalTo("/t/" + context.getContextTenant().getDomain() +
```

Comment:
it's better to remove the variable and use the actual value here to make it more readable. Same applies for all similar places.

Reply:
since the test is running on multi-tenant mode through data provider, the tenant domain is varying. Hence we need to use the tenant domain as a variable here.

Reply:
I'm referring to `key1Identifier` variable. It's hard to understand what we are selecting here.